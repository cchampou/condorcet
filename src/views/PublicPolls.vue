<template>
  <div v-if="polls">
    <div v-for="poll in polls" :key="poll.id">
      <h3>{{ poll.question }}</h3>
      <button @click="joinPoll(poll.id)">Rejoindre</button>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import { db } from "../firebase";

export default {
  name: "PublicPolls",
  data() {
    return {
      polls: null
    };
  },
  computed: {
    ...mapGetters("user", ["userId"])
  },
  async created() {
    let polls = [];

    const underPolls = await db
      .collection("polls")
      .where("isPublic", "==", true)
      .where("owner", ">", this.userId)
      .get();
    const overPolls = await db
      .collection("polls")
      .where("isPublic", "==", true)
      .where("owner", "<", this.userId)
      .get();
    underPolls.forEach(poll => {
      polls.push({ id: poll.id, ...poll.data() });
    });
    overPolls.forEach(poll => {
      polls.push({ id: poll.id, ...poll.data() });
    });

    this.polls = polls;
  },
  methods: {
    async joinPoll(id) {
      try {
        const users = (await db
          .collection("polls")
          .doc(id)
          .get()).data().users;

        db.collection("polls")
          .doc(id)
          .update({ users: [...users, this.userId] });
      } catch (error) {
        this.addNotification({
          status: "error",
          message: "Nous n'avons pas pu te faire rejoindre ce scrutin"
        });
      }
    }
  }
};
</script>

<style></style>
