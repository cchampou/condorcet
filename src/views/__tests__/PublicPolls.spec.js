import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import PublicPolls from "../PublicPolls.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("PublicPolls.vue", () => {
  it("should render", () => {
    const store = new Vuex.Store({
      modules: {
        user: {
          namespaced: true,
          getters: {
            userId: () => "ez9fze83n"
          }
        }
      }
    });

    const wrapper = shallowMount(PublicPolls, { localVue, store });

    expect(wrapper).toMatchSnapshot();
  });
});
