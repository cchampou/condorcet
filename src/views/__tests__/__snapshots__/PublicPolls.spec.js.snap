// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicPolls.vue should render 1`] = ``;
